src/rgb.rs

```diff
@@ -123,51 +122,182 @@ impl ANSIColorCode for Rgb {
     }
 }
 
-overload::overload!(
-    (lhs: ?Rgb) + (rhs: ?Rgb) -> Rgb {
-        Rgb::new(
-            lhs.r.saturating_add(rhs.r),
-            lhs.g.saturating_add(rhs.g),
-            lhs.b.saturating_add(rhs.b)
-        )
+fn rgb_add(lhs: &Rgb, rhs: &Rgb) -> Rgb {
+    Rgb::new(
+        lhs.r.saturating_add(rhs.r),
+        lhs.g.saturating_add(rhs.g),
+        lhs.b.saturating_add(rhs.b),
+    )
+}
+
+fn rgb_sub(lhs: &Rgb, rhs: &Rgb) -> Rgb {
+    Rgb::new(
+        lhs.r.saturating_sub(rhs.r),
+        lhs.g.saturating_sub(rhs.g),
+        lhs.b.saturating_sub(rhs.b),
+    )
+}
+
+fn rgb_mul_f32(lhs: &Rgb, rhs: &f32) -> Rgb {
+    Rgb::new(
+        (lhs.r as f32 * rhs.clamp(0.0, 1.0)) as u8,
+        (lhs.g as f32 * rhs.clamp(0.0, 1.0)) as u8,
+        (lhs.b as f32 * rhs.clamp(0.0, 1.0)) as u8,
+    )
+}
+
+fn f32_mul_rgb(lhs: &f32, rhs: &Rgb) -> Rgb {
+    Rgb::new(
+        (rhs.r as f32 * lhs.clamp(0.0, 1.0)) as u8,
+        (rhs.g as f32 * lhs.clamp(0.0, 1.0)) as u8,
+        (rhs.b as f32 * lhs.clamp(0.0, 1.0)) as u8,
+    )
+}
```

Comment:
This should be entirely commutative here

Reply:
oh yeah, of course.